cli/tests/activate.bats

```diff
@@ -2250,8 +2255,20 @@ EOF
 
   echo "$MANIFEST_CONTENTS" | "$FLOX_BIN" edit -d nested -f -
 
-  USER="$REAL_USER" run zsh --interactive --login -c "unset _already_ran_profile_common _already_ran_profile_zsh && expect $TESTS_DIR/activate/activate.exp nested"
+  # Unset the flags from `only-once` profile scripts because we expect them to
+  # be run once by the outer `zsh` command, then again by the inner `nested`
+  # profile scripts, and then no further.
+  FLOX_SHELL=zsh USER="$REAL_USER" NO_COLOR=1 run zsh --interactive --login -c \
+    "unset _already_ran_profile_common _already_ran_profile_zsh && expect $TESTS_DIR/activate/activate.exp nested"
   assert_success
+  # Outer in-place activation.
+  assert_output --partial - <<EOF
+sourcing hook.on-activate for first time
+sourcing profile.common for first time
+sourcing profile.zsh for first time
+EOF
+  # Inner interactive activation.
+  # We can't use `--partial - <<EOF` here for reasons that escape me.
```

Comment:
Are there escape characters in the `expect` output?

Reply:
Ah yes, `| cat -ev` shows carriage returns. I have no idea how to match that in BATS though 🙈 
```
sourcing hook.on-activate for first time$
   sourcing profile.common for first time$
   sourcing profile.zsh for first time$
   spawn /Users/dcarley/projects/flox/flox/cli/target/debug/flox activate --dir nested^M$
   M-bM- M-^A Preparing environment 'nested'...                                                                                                                                                                                       ^M^[[2KM-bM- M-^A Preparing environment 'nested'...                                                                                                                                                                                       ^M^[[2KM-bM-^\M-^E You are now using the environment 'nested'.^M$
   To stop using this environment, type 'exit'^M$
   ^M$
   exit^M$
   nested hook.on-activate^M$
   sourcing profile.common for first time^M$
   sourcing profile.zsh for first time^M$
   nested profile.common^M$
   nested profile.zsh^M$
   ^[[1m^[[7m%^[[27m^[[1m^[[0m                                                                                                                                                                                                                         ^M ^M^M^[[0m^[[27m^[[24m^[[Jflox [nested project-5] ^[[1m^[[0mdcarley@^[[4mmba15^[[24m> ^[[K^[[80C^[[32m/var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.T9DIYT/bats-run-aniObg/test/5/project-5^[[39m^[[178D^[[?2004he^Hexit^[[?2004l^M^M$
```

Reply:
I tried several variations of `\r` and literal `^M`, with both HEREDOC and `assert_line`, before giving up and piping the output to `tr -d '\r'`.